presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -170,21 +174,75 @@ public PartitionStatistics load(HivePartitionName key)
         partitionFilterCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize)
                 .build(asyncReloading(CacheLoader.from(this::loadPartitionNamesByFilter), executor));
 
-        partitionCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize)
-                .build(asyncReloading(new CacheLoader<HivePartitionName, Optional<Partition>>()
-                {
-                    @Override
-                    public Optional<Partition> load(HivePartitionName partitionName)
-                    {
-                        return loadPartitionByName(partitionName);
-                    }
+        if (isPartitioningEnabled) {
+            this.isPartitionVersioningEnabled = true;
+            long expire = 86400000;
 
-                    @Override
-                    public Map<HivePartitionName, Optional<Partition>> loadAll(Iterable<? extends HivePartitionName> partitionNames)
+            partitionStatisticsCache = newCacheBuilder(OptionalLong.of(expire), OptionalLong.empty(), 10000000)
+                    .build(asyncReloading(new CacheLoader<HivePartitionName, PartitionStatistics>()
                     {
-                        return loadPartitionsByNames(partitionNames);
-                    }
-                }, executor));
+                        @Override
+                        public PartitionStatistics load(HivePartitionName key)
+                        {
+                            return loadPartitionColumnStatistics(key);
+                        }
+
+                        @Override
+                        public Map<HivePartitionName, PartitionStatistics> loadAll(Iterable<? extends HivePartitionName> keys)
+                        {
+                            return loadPartitionColumnStatistics(keys);
+                        }
+                    }, executor));
+
+            partitionCache = newCacheBuilder(OptionalLong.of(expire), OptionalLong.empty(), 10000000)
+                    .build(asyncReloading(new CacheLoader<HivePartitionName, Optional<Partition>>()
+                    {
+                        @Override
+                        public Optional<Partition> load(HivePartitionName partitionName)
+                        {
+                            return loadPartitionByName(partitionName);
+                        }
+
+                        @Override
+                        public Map<HivePartitionName, Optional<Partition>> loadAll(Iterable<? extends HivePartitionName> partitionNames)
+                        {
+                            return loadPartitionsByNames(partitionNames);
+                        }
+                    }, executor));
+        }
+        else {
```

Comment:
I might have missed something, but what is the difference between the cache builders of these two branches?